src/lib/y2packager/product_finder.rb

```diff
@@ -148,10 +124,10 @@ def create_products(product_solvable, found_base_products, selected_base)
           base:         found_base_products.include?(product_name)
         }
 
-        list << ret.merge(product_data)
+        ret << data.merge(product_data)
```

Comment:
and when I see this data. Why not create that ProductDetails and return just object?